### PR TITLE
Run validation with updated resource values

### PR DESCRIPTION
**Pull request by rottencandy**
## Fixes 
https://issues.redhat.com/browse/HAC-1286

## Description
Bug in formik causes validation to happen on the previous value instead of the updated value when `setFieldTouched` is used.

I tried all workarounds from https://github.com/jaredpalmer/formik/issues/2083 but this is the only one that worked.

## Type of change

- [ ] Feature
- [x] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

https://user-images.githubusercontent.com/20013884/177797951-40a04ef8-b43f-4da8-817c-9c4f937522ca.mp4

## How to test or reproduce?
Try to add negative values in resource limit field.

## Browser conformance: 

- [ ] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: src/shared/components/formik-fields/ResourceLimitField.tsx [hunk 1]
```diff
@@ -34,6 +35,8 @@ const ResourceLimitField: React.FC<ResourceLimitFieldProps> = ({
           setFieldValue(props.name, val.value);
           setFieldTouched(props.name, true);
           setFieldValue(unitName, val.unit);
+          // run validation manually so that formik uses the latest value
+          validateForm(set(values, props.name, val.value));
```
[the remark this answers is not in the document]
Comment (rottencandy) [reply src/shared/components/formik-fields/ResourceLimitField.tsx:39]: Tried `validateField`, it's still using the old value.

File: src/shared/components/formik-fields/ResourceLimitField.tsx [hunk 2]
```diff
@@ -31,6 +32,8 @@ const ResourceLimitField: React.FC<ResourceLimitFieldProps> = ({
       <RequestSizeInput
         {...props}
         onChange={(val) => {
+          // run validation manually so that formik uses the latest value
+          validateForm(set(values, props.name, val.value));
           setFieldValue(props.name, val.value);
```
Comment (rohitkrai03) [inline src/shared/components/formik-fields/ResourceLimitField.tsx:37]: Have you tried this workaround? I tried it and it works for me.
```suggestion
         setTimeout(() => setFieldValue(props.name, val.value));
```

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: src/shared/components/formik-fields/ResourceLimitField.tsx (hunk 2 above)
Comment (rottencandy) [reply src/shared/components/formik-fields/ResourceLimitField.tsx]: Thanks! Updated

**Review by rohitkrai03 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**